datafusion/physical-plan/src/insert.rs

```diff
@@ -151,6 +162,16 @@ impl FileSinkExec {
         }
         Ok(streams)
     }
+
+    /// Returns insert sink
+    pub fn sink(&self) -> &dyn DataSink {
+        self.sink.as_ref()
+    }
+
+    /// Returns the metrics of the underlying [DataSink]
+    pub fn metrics(&self) -> Option<MetricsSet> {
+        self.sink.metrics()
```

Comment:
Ahh this was the wrong place, it should be in `impl ExecutionPlan for FileSinkExec`